packages/core/src/UserState.ts

```diff
@@ -9,26 +10,29 @@ import {
     ZkIdentity,
 } from '@unirep/crypto'
 import { Circuit, genProofAndPublicSignals } from '@unirep/circuits'
+import { IAttestation, Attestation } from '@unirep/contracts'
+import { NUM_ATTESTATIONS_PER_PROOF } from '@unirep/config'
+
 import {
     defaultUserStateLeaf,
     genEpochKey,
     genNewSMT,
     genEpochKeyNullifier,
     genReputationNullifier,
+    stringifyAttestation,
 } from './utils'
-import { IAttestation, IUnirepState, UnirepState } from './UnirepState'
-import { NUM_ATTESTATIONS_PER_PROOF } from '@unirep/config'
+import { IUnirepState, UnirepState } from './UnirepState'
 
 interface IUserStateLeaf {
     attesterId: BigInt
     reputation: Reputation
 }
 
 interface IReputation {
-    posRep: BigInt
-    negRep: BigInt
-    graffiti: BigInt
-    signUp: BigInt
+    posRep: BigNumber
```

Comment:
Shall we about creating a new module(file) for `Reputation`? I think the project structure will be a little bit easier to read.